### PR TITLE
Add support for log enricher profile recording

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Implements profile recording by using the log enricher.

It also fixes a couple of issues with the log enricher:

- We store the `containerID` here, which should be `rawContainerID`: https://github.com/kubernetes-sigs/security-profiles-operator/blob/cd71f464ca8606ff427bf8a2826c682d92053a28/internal/pkg/daemon/enricher/container.go#L100
- We re-init the local `containers` for every loop: https://github.com/kubernetes-sigs/security-profiles-operator/blob/cd71f464ca8606ff427bf8a2826c682d92053a28/internal/pkg/daemon/enricher/container.go#L66
- We can completely remove the `containers` map and only rely on the cache to stay safe: https://github.com/kubernetes-sigs/security-profiles-operator/blob/cd71f464ca8606ff427bf8a2826c682d92053a28/internal/pkg/daemon/enricher/container.go#L58

#### Which issue(s) this PR fixes:

PoC of https://github.com/kubernetes-sigs/security-profiles-operator/issues/510

#### Does this PR have test?

None

#### Special notes for your reviewer:

Possible follow-upds:
- Update the docs to cover the new feature and `ProfileRecording` field
- Use GRPC streaming for the syscalls (open a stream on the beginning of the enricher and use that one for pushing the syscalls)

#### Does this PR introduce a user-facing change?

```release-note
Added support for recording profiles by using the log enricher.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
I'll update the docs in a follow-up PR.

File: internal/pkg/daemon/enricher/container.go [hunk 1]
```diff
@@ -111,12 +114,13 @@ func (e *Enricher) getContainerInfo(
...
-	if !found {
-		return nil, errors.Wrap(err, "no container info for container ID")
+	if info, err := e.infoCache.Get(
+		containerID,
+	); !errors.Is(err, ttlcache.ErrNotFound) {
+		return info.(*containerInfo), nil
 	}
 
-	return info, err
+	return nil, errors.New("found nothing")
```
Comment (JAORMX) [inline internal/pkg/daemon/enricher/container.go:123]: `found nothing` isn't a very descriptive error message, we probably want to keep using something along the lines of `"no container info for container ID"`

Comment (saschagrunert) [reply internal/pkg/daemon/enricher/container.go]: Fixed :+1:

**Review by JAORMX — APPROVED** [review]
/lgtm

**Pull request merged**